cmd/gaia/cli_test/cli_test.go

```diff
@@ -64,6 +64,15 @@ func TestGaiaCLISend(t *testing.T) {
 	assert.Equal(t, int64(20), barAcc.GetCoins().AmountOf("steak").Int64())
 	fooAcc = executeGetAccount(t, fmt.Sprintf("gaiacli account %v %v", fooCech, flags))
 	assert.Equal(t, int64(30), fooAcc.GetCoins().AmountOf("steak").Int64())
+
+	// test memo
+	executeWrite(t, fmt.Sprintf("gaiacli send %v --amount=10steak --to=%v --name=foo --memo 'testmemo'", flags, barCech), pass)
+	tests.WaitForNextHeightTM(port)
+
+	barAcc = executeGetAccount(t, fmt.Sprintf("gaiacli account %v %v", barCech, flags))
+	assert.Equal(t, int64(30), barAcc.GetCoins().AmountOf("steak").Int64())
+	fooAcc = executeGetAccount(t, fmt.Sprintf("gaiacli account %v %v", fooCech, flags))
+	assert.Equal(t, int64(20), fooAcc.GetCoins().AmountOf("steak").Int64())
```

Comment:
Not sure if possible through just cli commands, but ideally this should check that the memo field is in history. Currently this is just checking that inclusion of the memo field doesn't break sends.

Reply:
Checking memo in history would be nice but is inconvenient to do through CLI tests, ideally in the future we'll have better Tendermint integration tests - https://github.com/cosmos/cosmos-sdk/issues/1313.